### PR TITLE
ARROW-10733: [R] Improvements to Linux installation troubleshooting

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by nealrichardson** [comment]
Passing builds on my fork: https://github.com/nealrichardson/arrow/actions/runs/449373084

File: r/vignettes/install.Rmd [hunk 1]
```diff
@@ -175,25 +181,24 @@ tune one of several parameters. Here are some known complications and ways to ad
...
-on all Linux distributions, so if you see this message because the C++ libraries
-errored while building,
+compatible with the current version of the R package.
+
+It is expected that C++ dependencies should be built successfully
+on all Linux distributions, so you should not see this message. If you do,
+please check the "Known installation issues" below to see if any apply.
+If none apply, retry the installation with `arrow::install_arrow(verbose = TRUE)`
+so that details on what failed are shown, then
 please [report an issue](https://issues.apache.org/jira/projects/ARROW/issues).
```
Comment (jonkeane) [inline r/vignettes/install.Rmd:196]: Do we want to proactively ask for the install log in the report? I've seen a few jiras where that's usually the first ask.

File: r/vignettes/install.Rmd [hunk 2]
```diff
@@ -286,12 +291,12 @@ so that we can attempt to improve the script.
...
 the environment variables `CC` and `CXX` either in the shell or in R's `Makeconf`.
 For CentOS 7 and above, both the Arrow system packages and the C++ binaries
-for R are built with the default system compilers. If you want to use either of these 
-and you have a `devtoolset` installed, set `CC=/usr/bin/gcc CXX=/usr/bin/g++` 
-to use the system compilers instead of the `devtoolset`. 
-Alternatively, if you want to build `arrow` with the newer `devtoolset` compilers, 
+for R are built with the default system compilers. If you want to use either of these
+and you have a `devtoolset` installed, set `CC=/usr/bin/gcc CXX=/usr/bin/g++`
+to use the system compilers instead of the `devtoolset`.
+Alternatively, if you want to build `arrow` with the newer `devtoolset` compilers,
```
Comment (jonkeane) [inline r/vignettes/install.Rmd:297]: 👍 this all looks good and matches what I was experiencing on centos

**Comment by jonkeane** [comment]
This looks good, one minor addition/comment.

File: r/vignettes/install.Rmd (hunk 1 above)
Comment (nealrichardson) [reply r/vignettes/install.Rmd:196]: Yeah the first ask is almost always to rerun with verbosity (what I added in this PR) and share the logs. Probably would be good to explicitly ask for the full output here.

**Pull request closed without merging**